### PR TITLE
5. Split and convert Utils.kt

### PR DESCRIPTION
**Pull request by yiqunw700**
MathUtils calls one Utils api. Double check when #1916 merged.

### CODE REVIEW

**Review by timothyfroehlich — APPROVED** [review]
Looks good. I wonder if proto's merge functions would be cleaner than our manual merges, but that's not worth taking the time to do right now.

File: designcompose/src/main/java/com/android/designcompose/utils/GraphicUtils.kt [hunk 1]
```diff
@@ -0,0 +1,530 @@
...
+                    scaleMode = backgroundImage.scaleMode,
+                ),
+                backgroundImage.opacity,
+            )
+        }
+        return null
+    }
+
+    if (hasLinearGradient()) {
+        val linearGradient = getLinearGradient()
```
[the remark this answers is not in the document]
Comment (yiqunw700) [reply designcompose/src/main/java/com/android/designcompose/utils/GraphicUtils.kt:434]: Done.

File: designcompose/src/main/java/com/android/designcompose/utils/Utils.kt [hunk 2]
```diff
@@ -358,98 +227,140 @@ internal fun com.android.designcompose.serdegen.Path.log() {
...
-            if (textArray.size == styledTextData.value.styled_texts.size) {
+            if (textArray.size == styledTextData.styledTextsCount) {
                 val output = mutableListOf<StyledTextRun>()
                 for (i in textArray.indices) {
                     output.add(
-                        StyledTextRun(textArray[i], styledTextData.value.styled_texts[i].style)
+                        StyledTextRun.newBuilder()
+                            .setText(textArray[i])
+                            .setStyle(styledTextData.getStyledTexts(i).style)
+                            .build()
```
[the remark this answers is not in the document]
Comment (yiqunw700) [reply designcompose/src/main/java/com/android/designcompose/utils/Utils.kt:363]: Done.

File: designcompose/src/main/java/com/android/designcompose/utils/Utils.kt [hunk 3]
```diff
@@ -458,20 +369,20 @@ internal fun getTextContent(
...
+        if (styledTextData.styledTextsCount == 1) {
             val strResId = context.resources.getIdentifier(resName, "string", context.packageName)
             if (strResId != Resources.ID_NULL) {
                 Log.w(TAG, "Single style found, fallback to string resource")
                 return mutableListOf(
-                    StyledTextRun(
-                        context.getString(strResId),
-                        styledTextData.value.styled_texts[0].style,
-                    )
+                    StyledTextRun.newBuilder()
```
[the remark this answers is not in the document]
Comment (yiqunw700) [reply designcompose/src/main/java/com/android/designcompose/utils/Utils.kt:377]: Done.

File: designcompose/src/main/java/com/android/designcompose/utils/ViewStyleUtils.kt [hunk 4]
```diff
@@ -16,363 +16,367 @@
...
-    val style = ViewStyle.Builder()
-    val nodeStyle = NodeStyle.Builder()
-    val layoutStyle = LayoutStyle.Builder()
-    nodeStyle.text_color =
-        if (
-            override.nodeStyle.text_color.getOrNull()?.getType() ==
-            BackgroundType.None(com.novi.serde.Unit())
-        ) {
-            override.nodeStyle.text_color
+    val style = ViewStyle.newBuilder()
```
[the remark this answers is not in the document]
Comment (yiqunw700) [reply designcompose/src/main/java/com/android/designcompose/utils/ViewStyleUtils.kt]: Done.

**Comment by yiqunw700** [comment]
> Looks good. I wonder if proto's merge functions would be cleaner than our manual merges, but that's not worth taking the time to do right now.

Added a TODO. I thought about using proto's merge functions.But 
1. I am not clear how the merge is done in proto
2. the default values are actually from the rust code, so they may not equal to the default value in the generated proto.

Let me file a ticket to revisit this merge function after the conversion is done.

**Pull request merged**